connector-packager/connector_packager/xsd_validator.py

```diff
@@ -163,6 +163,21 @@ def validate_file_specific_rules(file_to_test: ConnectorFile, path_to_file: Path
                                                  " already exists. Cannot have multiple fields with the same name.")
 
                     return False
+                if field_name == 'instanceurl':
+                    used_for_oauth = False
+                    for conditions in child.iter("conditions"):
+                        for condition in conditions.iter("condition"):
+                            if 'field' in condition.attrib:
```

Comment:
I don't like this any better.
```suggestion
                            for conditions in child.iter("conditions"):
                        for condition in conditions.iter("condition"):
                            if not used_for_oauth:
                                used_for_oauth = ('field' in condition.attrib) & \
                                                    (condition.attrib['field'] == 'authentication') & \
                                                    ('value' in condition.attrib) & \
                                                    (condition.attrib['value'] == 'oauth')
```